app/controllers/answers_controller.rb

```diff
@@ -0,0 +1,28 @@
+post '/answers' do
+
+  choice = Choice.find(params[:choice])
+  survey = choice.question.survey
+  this_question = choice.question
+  questions = survey.questions
+
+  Answer.create(taker_id: current_user.id, choice_id: params[:choice], question_id: this_question.id, survey_id: survey.id)
+
+  if this_question == survey.questions.last 
+    if request.xhr?
+      @survey = survey
+      erb :"/surveys/stats", layout: false
+    else
+     redirect "/surveys/#{survey.id}/stats"
+    end
+  else
+    # @question = Question.find(question.id+1)
+    index = questions.find_index(this_question)+1
```

Comment:
`answer.next_question_in_survey`

This method would look something like

`self.survey.next_question(answer.question)`

Something like this at least.